airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts

```diff
@@ -0,0 +1,22 @@
+import { apiOverride } from "core/request/apiOverride";
+
+export interface RedirectUrlResponse {
+  redirectUrl?: string;
+}
+
+// eslint-disable-next-line
+type SecondParameter<T extends (...args: any) => any> = T extends (config: any, args: infer P) => any ? P : never;
+
+/**
+ * Return a JSON datastructure that contains the URL that should be redirected to in the redirectUrl field.
+ * @summary Get the Speakeasy Callback URL
+ */
+export const getSpeakeasyCallbackUrl = (options?: SecondParameter<typeof apiOverride>, signal?: AbortSignal) => {
+  return apiOverride<RedirectUrlResponse>({ url: `/speakeasy_callback_url`, method: "get", signal }, options);
+};
+
+type AwaitedInput<T> = PromiseLike<T> | T;
+
+type Awaited<O> = O extends AwaitedInput<infer T> ? T : never;
+
+export type GetSpeakeasyCallbackUrlResult = NonNullable<Awaited<ReturnType<typeof getSpeakeasyCallbackUrl>>>;
```

Comment:
This is the orval-generated API code. The Open API spec lives in `airbyte-cloud`, so I think it's more straightforward to commit directly here.

Reply:
I've moved it for now into a `speakeasy` folder. Since I think we should keep the general cleanup (and merging those APIs as you suggested, which I also believe is the better actual solution) into separate cleanup.

Reply:
I think we're mixing up terms here - I intended `packages/cloud/lib/domain/cloudApi/api.ts` to be scoped to all the [Public Cloud API](https://docs.google.com/document/d/1FjYyeSqCGcUW2EO0PJ5p7j5afsodM9GLSI3WBwtQHKo/edit) endpoints. This is a new service that currently only does the speakeasy redirect, which is why that's the only endpoint there. I agree that the naming is confusing though, so I'm happy to revisit this!